app.js

```diff
@@ -48,49 +50,52 @@ app.post('/auth', jwtAuth);
 router.use('/api', verifyJwt);
 /* end jsonwebtoken for authentication */
 
-function createProduct({ name, reviews = [] }) {
-  const product = new Product(name);
-  if (reviews instanceof Array && reviews.length > 0) {
-    reviews.forEach(review => {
-      product.addReview(review);
-    });
-  }
-  products.push(product);
-  return product;
-}
-
 router.route('/api/products')
-  .get((request, response) => {
+  .get(async (request, response) => {
+    let products;
+    try {
+      products = await DB.getProducts();
+    }
+    catch(e) {
+      response.json({ error: e });
+    }
     response.json(products);
   })
-  .post((req, res) => {
-    const newProduct = createProduct(req.body);
+  .post(async (req, res) => {
+    const { name } = req.body;
+    const product = {
+      name
+    };
+    let newProduct;
+    try {
+      newProduct = await DB.createProduct(product);
+    }
+    catch(e) {
+      res.send({ error: "product can't be created" });
+    }
     res.send(newProduct);
   });
 
-function getProduct(id) {
-  const product = getModelById(products, id);
-  if (product) {
-    return product;
+router.get('/api/products/:id', async (request, response) => {
```

Comment:
It doesn't work. Probably because you don't cast string to number.

Reply:
Thx! fixed. The problem was wrong naming in static async getProductById.